oidc/TokenResponse.ml

```diff
@@ -76,3 +77,16 @@ let validate ?clock_tolerance ?nonce ~jwks ~(client : Client.t)
         |> Result.map (fun _ -> t)
       | None -> Error (`Msg "Could not find JWK"))
   | Error e -> Error e
+
+let to_json { scope; expires_in; access_token; refresh_token; id_token; _ } =
```

Comment:
Should we make id_token a Jose.Jwt.t?
Or if we have something in oidc that wraps it that should be used. Since id_token is always a jwt anyway.

Reply:
as per https://darutk.medium.com/diagrams-of-all-the-openid-connect-flows-6968e3990660:

> When the value of response_type is code, but if openid is not included in the scope request parameter, the request is just an [authorization code flow](https://tools.ietf.org/html/rfc6749#section-4.1) which is defined in RFC 6749. On the other hand, if openid is included in the scope request parameter, an ID token is issued from the token endpoint in addition to an access token.

It seems like `id_token` isn't included everytime, so if we force it to be a JWT we should also make it optional?

Reply:
I'm inclined to merge this as is since the change is just better than what we have now. But as we spoke about offline making `id_token` optional and folding oauth2 functionality into the oidc library would be the best.
Do you want to do this or should I merge and open a issue?